### PR TITLE
Generates winget manifest during GitHub Actions

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This generates winget config directly instead of relying on winget-create. Notably, this covers sha-256 and product-code aspects:

> Package installers in MSI format use [Product Codes](https://docs.microsoft.com/en-us/windows/win32/msi/product-codes) to uniquely identify applications. The product code for a given version of a package should be included in the manifest to help ensure the best **upgrade** experience.

This tests that we can extract on every commit that changes the MSI script. It also extracts logic so that less new code is executed on release.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
found another way to get this (I think) hold on

**Comment by codefromthecrypt** [comment]
nah false alarm. I was hoping `file /path/to/func-e.msi` had the same ID in it, but that's not the case.

File: packaging/msi/verify_msi.cmd [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+:: distributed under the License is distributed on an "AS IS" BASIS,
+:: WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+:: See the License for the specific language governing permissions and
+:: limitations under the License.
+
+:: verify_msi is written in cmd because msiexec doesn't agree with git-bash
+:: See https://github.com/git-for-windows/git/issues/2526
+@echo off
+if not defined MSI_FILE set MSI_FILE=dist\func-e_windows_amd64\func-e.msi
+:: this is how you do MSI_PRODUCT_CODE=$(command) in cmd:
```
Comment (codefromthecrypt) [inline packaging/msi/verify_msi.cmd:19]: I put this in originally to verify it works. I'll find another place to do verification which isn't so crufty. probably set GH env in msi.yaml (using powershell). After that, this cmd file will be easier on the eyes

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
found another way 💯 this works on linux and darwin as long as GNOME msitools are installed

```bash
$ msiinfo export  dist/func-e_windows_amd64/func-e.msi Property
Property	Value
s72	l0
Property	Property
ARPURLINFOABOUT	https://func-e.io
ARPNOREPAIR	1
ARPNOMODIFY	1
ARPPRODUCTICON	icon.ico
setx	setx.exe
Manufacturer	Tetrate
ProductLanguage	1033
ProductCode	{40822491-10DE-4D1C-AE0C-53911FC1FFA9}
ProductName	func-e
ProductVersion	0.0.1
UpgradeCode	{FC6C326B-1DB7-4D48-8F05-E13E79DF4D0E}
SecureCustomProperties	NEWERVERSIONDETECTED;OLDERVERSIONBEINGUPGRADED
```

**Comment by codefromthecrypt** [comment]
I rewrote this to also validate the winget yaml ( on all supported dev operating systems )

File: .github/workflows/msi.yaml [hunk 2]
```diff
@@ -70,19 +70,32 @@ jobs:
...
+
+      # This tests writing the manifest via GHA env works. Validation requires too much setup.
+      # See https://github.com/microsoft/winget-cli/issues/754#issuecomment-896475895
+      - name: "Write winget manifest"
+        run: |
+          manifest_path=./Tetrate.func-e.yaml
+          cat > "${manifest_path}" <<'EOF'
+          ${{ env.WINGET_MANIFEST }}
+          EOF
+          yamllint -sd '{extends: default, rules: {line-length: disable}}' Tetrate.func-e.yaml
```
Comment (mathetake) [inline .github/workflows/msi.yaml:101]: yamllint on windows! 😮

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
Thanks for taking a look @mathetake!

**Pull request merged**